### PR TITLE
Persistence component testsuite

### PR DESCRIPTION
**Pull request by lailabougria**

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
LOL, CLA assistant go home, you're drunk

**Comment by timbussmann** [comment]
I reran the CLA assistant, seems to be sober again today 👍

File: src/NServiceBus.AcceptanceTesting/NServiceBus.AcceptanceTesting.csproj [hunk 1]
```diff
@@ -20,4 +20,8 @@
     <Description>Acceptance testing framework for NServiceBus endpoints. This is an unsupported package.</Description>
   </PropertyGroup>
 
+  <ItemGroup>
+    <ProjectCapability Remove="TestContainer" />
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTesting/NServiceBus.AcceptanceTesting.csproj:24]: that seems like some accident we should revert?

Comment (bording) [reply src/NServiceBus.AcceptanceTesting/NServiceBus.AcceptanceTesting.csproj:24]: No, this was my intentional change that I was talking about in the last sync meeting.

Comment (timbussmann) [reply src/NServiceBus.AcceptanceTesting/NServiceBus.AcceptanceTesting.csproj:24]: ah ok, I missed that the element name was `ProjectCapability`. Ok that makes sense. Maybe we can add a comment to clarify a little bit why this was added?

File: src/NServiceBus.PersistenceTesting/Sagas/When_persisting_the_same_saga_twice_in_two_sessions_on_the_same_thread.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+{
+    using System;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using NUnit.Framework;
+    using Persistence;
+
+    public class When_persisting_the_same_saga_twice_in_two_sessions_on_the_same_thread : SagaPersisterTests
+    {
+        [Test] // TODO: Do we need this? What's the use of this test as opposed to When_retrieving_same_saga_on_the_same_thread?
```
Comment (bording) [inline src/NServiceBus.PersistenceTesting/Sagas/When_persisting_the_same_saga_twice_in_two_sessions_on_the_same_thread.cs:11]: This test still has a TODO that needs to be evaluated.

File: src/NServiceBus.PersistenceTesting/IPersistenceTestsConfiguration.cs [hunk 3]
```diff
@@ -34,6 +34,10 @@ public interface IPersistenceTestsConfiguration
 
         SagaMetadataCollection SagaMetadataCollection { get; }
 
+        /// <summary>
+        /// Initialise any properties needed within the Configure method
+        /// </summary>
+        /// <returns></returns>
```
Comment (timbussmann) [inline src/NServiceBus.PersistenceTesting/IPersistenceTestsConfiguration.cs:40]:
```suggestion
```

Comment (bording) [reply src/NServiceBus.PersistenceTesting/IPersistenceTestsConfiguration.cs:40]: I don't see value in having this one XML comment at all. I think it should be removed entirely.

**Pull request merged**